src/phpDocumentor/Transformer/Writer/Twig/FlySystemLoader.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link http://phpdoc.org
+ */
+
+namespace phpDocumentor\Transformer\Writer\Twig;
+
+use League\Flysystem\FileNotFoundException;
+use League\Flysystem\FilesystemInterface;
+use Twig\Error\LoaderError;
+use Twig\Loader\LoaderInterface;
+use Twig\Source;
+use function rtrim;
+use function sprintf;
+
+final class FlySystemLoader implements LoaderInterface
+{
+    /** @var FilesystemInterface */
+    private $filesystem;
+
+    /** @var string */
+    private $templatePath;
+
+    public function __construct(FilesystemInterface $filesystem, string $templatePath = '')
+    {
+        $this->filesystem = $filesystem;
+        $this->templatePath = $templatePath;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getSourceContext($name)
+    {
+        $this->guardTemplateExistsAndIsFile($name);
+
+        $path = $this->resolveTemplateName($name);
+
+        return new Source(
+            $this->filesystem->read($path),
+            $name,
+            $path
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists($name)
+    {
+        return $this->filesystem->has($this->resolveTemplateName($name));
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getCacheKey($name)
+    {
+        $this->guardTemplateExistsAndIsFile($name);
+
+        return $name;
```

Comment:
This just returns the unaltered string `$name` that was passed in?  Is the actual effect on this method call to do some state change in the background? At first glance, I thought maybe it should be a `void` return instead.

Or, is some method chaining/composing done in a way that a method wrapped around this method at runtime is also needing `$name`, and that's why it is returned like this?

Reply:
The main effect of this function is the guard; if that throws a loader error than Twig knows the cache key cannot be set. And the template name is a valid cache key in this case :)

Reply:
So, should it be a `void` return then?  The fact that it returns `$name` seems significant... and then the next realization that it returns the same `$name` _unaltered_ seems _extremely_ significant.  It's the kind of thing that, if I was in this code troubleshooting something, seeing this return done this way would make me sit up and take notice... and then go digging around calling code to try to understand its rationale, trying to understand why it was _important_ to return the same `$name` value that was passed in as an arg.  If there is so true significance for returning it that way, can it be `void` instead?

Reply:
This is a requirement of the base Twig Loader; not much I can do there I am afraid

Reply:
Ah hah, there's the unseen significance... that might be worth an inline comment, or even an `@internal` tag, so that a later refactoring won't see the same thing I did and think "that's not necessary... let's just `void` this instead".

Reply:
Done and pushed to master